.github/workflows/solutions_ci.yml

```diff
@@ -0,0 +1,43 @@
+name: CI for SAF templates
+on:
+  pull_request:
```

Comment:
Pull requests against branches other than `main` are rare, but I want to make sure that is your intent here. Without a branch filter, this will apply to all PRs.

Reply:
@nick-marnik Yes, it is intentional. Given that the tests will only run if changes to the solution templates have been made, I did not see the reason to also restrict branches. But please, let me know if you think there are reasons to add the filter!

Reply:
The permissions on this repository are a bit unique, but for other private repos, I think including a `branches: main` filter makes sense because:
- branch protections are generally only configured for main
- if I have write access to a repo, I generally wouldn't use a PR to merge changes from `branchA` into `branchB` - I would simply do the merge myself and then push directly to `branchB` without a PR
- we generally don't use long-lived feature branches, so this would be an extremely rare operation
- if a e2e workflow is triggered for a feature branch, it may cause a failure and raise flags for investigation when it isn't necessary

It's not a hill I want to die on - the way it is won't hurt anything. I leave it up to you to determine if the filter is meaningful to add.